dleq_vrf/src/keys.rs

```diff
@@ -86,7 +84,7 @@ pub struct SecretKey<K: AffineRepr> {
     pub(crate) thin: ThinVrf<K>,
 
     /// Secret key represented as a scalar.
-    pub(crate) key: SecretScalar<<K as AffineRepr>::ScalarField>,
+    pub(crate) key: SecretScalar<K::ScalarField>,
```

Comment:
Is this working purely because `<K: AffineRepr>` exists in the struct?  Is this a new rust feature?

Reply:
`K` is bound only by the `AffineRepr` trait. There are no ambiguities for the `ScalarField` associated type